src/Microsoft.AspNet.Hosting/Internal/WebApplication.cs

```diff
@@ -98,14 +127,14 @@ public virtual IApplication Start()
             _applicationLifetime.NotifyStarted();
             logger.Started();
 
-            return new Application(ApplicationServices, Server.Features, new Disposable(() =>
-            {
-                logger.Shutdown();
-                _applicationLifetime.StopApplication();
-                Server.Dispose();
-                _applicationLifetime.NotifyStopped();
-                (_applicationServices as IDisposable)?.Dispose();
-            }));
+            return new Disposable(() =>
```

Comment:
I don't like that. Disposable seems fine. I'm not sure why you;d need anything else.